hazelcast/src/test/java/com/hazelcast/logging/Slf4jLoggerLevelChangeTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.logging;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.logging.impl.LoggingServiceImpl;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.IsolatedLoggingRule;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.logging.Level;
+
+import static com.hazelcast.test.IsolatedLoggingRule.LOGGING_TYPE_SLF4J;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class Slf4jLoggerLevelChangeTest extends HazelcastTestSupport {
+
+    @Rule
+    public final IsolatedLoggingRule isolatedLoggingRule = new IsolatedLoggingRule();
+
+    private LoggingServiceImpl loggingService;
+
+    @Before
+    public void before() {
+        isolatedLoggingRule.setLoggingType(LOGGING_TYPE_SLF4J);
+        HazelcastInstance instance = createHazelcastInstance();
+        loggingService = (LoggingServiceImpl) instance.getLoggingService();
+    }
+
+    @Test(expected = HazelcastException.class)
+    public void test() {
+        // slf4j is not supported currently
```

Comment:
slf4j does not support changing log level at runtime?

Reply:
Unfortunately, there is no public or internal API for changing the level in slf4j, so for each logging backend supported by slf4j a separate specialized code should be written to extract the loggers from internal slf4j wrappers. The problem is that the wrappers are internal and specific for every logging backend, so they are not necessarily stable across versions of slf4j. In theory, we could support some set of popular logging backends for slf4j, but that would require inspecting at least recent versions of slf4j and wrappers to understand how stable the implementation of each wrapper is and how we could access it, regularly or through reflection. After discussion with Jiri and Matko we decided to postpone that.